File: app/src/main/java/com/jerson/hcdc_portal/database/AccountDao.java
package com.jerson.hcdc_portal.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.jerson.hcdc_portal.model.AccountModel;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;

@Dao
public interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertAccount(List<AccountModel> list);

    @Query("select * from account")
    Flowable<List<AccountModel>> getAccounts();


    @Query("delete from account")
    Completable deleteAccount();
}

File: app/src/main/java/com/jerson/hcdc_portal/data/local/dao/AccountDao.kt
package com.jerson.hcdc_portal.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.jerson.hcdc_portal.domain.model.Account
import kotlinx.coroutines.flow.Flow

@Dao
interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsertAccount(accounts: List<Account>)

    @Query("delete from accounts")
    suspend fun deleteAllAccounts()

    @Query("delete from accounts where termId = :termId")
    suspend fun deleteAccount(termId:Int)

    @Query("select * from accounts")
    fun getAccounts(): Flow<List<Account>>

    @Query("select * from accounts where termId = :termId")
    fun getAccounts(termId:Int): Flow<List<Account>>
}